cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -272,6 +277,82 @@ def update_from_dict(cls, key_values, descriptor, user, save=True):
+        :return: a list of error objects
+        """
+        errors = []
+        teams_configuration_model = settings_dict.get('teams_configuration', {})
+        if teams_configuration_model == {}:
+            return errors
+        proposed_topics = teams_configuration_model.get('value').get('topics')
+
+        proposed_max_team_size = teams_configuration_model.get('value').get('max_team_size')
+        if proposed_max_team_size <= 0:
+            message = 'Team size must be greater than zero.'
+            errors.append({'key': 'team_settings', 'message': message, 'model': teams_configuration_model})
+
+        proposed_topic_ids = [proposed_topic['id'] for proposed_topic in proposed_topics]
+        proposed_topic_dupe_ids = {x for x in proposed_topic_ids if proposed_topic_ids.count(x) > 1}
+        if len(proposed_topic_dupe_ids) > 0:
+            message = 'The following ids are duplicated: ' + ','.join(proposed_topic_dupe_ids)
+            errors.append({'key': 'team_settings', 'message': message, 'model': teams_configuration_model})
+
+        for proposed_topic in proposed_topics:
+            topic_validation = cls.validate_single_topic(proposed_topic)
+            if topic_validation:
+                topic_validation['model'] = teams_configuration_model
+                errors.append(topic_validation)
+
+        return errors
+
+    @classmethod
+    def validate_single_topic(cls, topic_settings):
+        """
+        Helper method that validates a single teamset setting.
+        The following conditions result in errors:
+        > unrecognized extra keys
+        > max_team_size <= 0
+        > no name, id or description property
+        > unrecognized teamset type
+        :param topic_settings: the proposed setting beingi validated
+        :return: an error object if error exists, otherwise None
+        """
+        error_list = []
+        valid_teamset_types = ['public_managed', 'private_managed', 'open']
+        valid_keys = {'id', 'name', 'description', 'max_team_size', 'type'}
+        teamset_type = topic_settings.get('type', {})
+        if teamset_type:
+            if teamset_type not in valid_teamset_types:
+                error_list.append('Unknown teamset type.')
+        max_team_size = topic_settings.get('max_team_size', {})
+        if max_team_size and max_team_size <= 0:
+            error_list.append('Team size must be greater than zero.')
+        teamset_id = topic_settings.get('id', {})
+        if not teamset_id:
+            error_list.append('id attribute must not be empty.')
+        teamset_name = topic_settings.get('name', {})
+        if not teamset_name:
+            error_list.append('name attribute must not be empty.')
+        teamset_desc = topic_settings.get('description', {})
+        if not teamset_desc:
+            error_list.append('description attribute must not be empty.')
+
+        keys = set(topic_settings.keys())
+        key_difference = keys - valid_keys
```

Comment:
The fact you can do this in Python is nifty.

Reply:
👍